### PR TITLE
Swift 3 naming guidelines

### LINKED ISSUE #130: Update APIs to follow swift 3 naming guidelines
https://swift.org/blog/swift-3-api-design/
- [x] Update public API (not private for the moment)
- [x] Update example app and playground
- [x] Update README code
- [x] Write changelog with BCHs
- [x] PR
- [ ] Release 2.0.0 📦

### PR DESCRIPTION
**Pull request by joanromano**
Closes #130

### CODE REVIEW

File: Source/Router.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ public struct Request {
 /**
  A protocol to adopt when a `Serializable object needs to also provide response status code and/or headerFields
  For example you may use `Response` to wrap your `Serializable` object to just achieve the result or directly implement the protocol.
- For example `JSONAPISerializer` implement the protocol in order to be able to provide custom status code in the response.
+ For example `JSONAPIError` implement the protocol in order to be able to provide custom status code in the response.
```
Comment (MP0w) [inline Source/Router.swift:39]: Good catch

File: Source/Serializer.swift [hunk 2]
```diff
@@ -50,18 +50,50 @@ public extension Serializable {
...
-        guard let object = serialize() else { return nil }
+        guard let object = serialized() else { return nil }
         
         if !JSONSerialization.isValidJSONObject(object) {
             return nil
         }
         return try? JSONSerialization.data(withJSONObject: object, options: .prettyPrinted)
     }
+    
+    fileprivate func serializeObject(_ value: Any, keyTransformer: KeyTransformer?) -> Any? {
```
Comment (MP0w) [inline Source/Serializer.swift:61]: Isn't private enough?

**Review by MP0w — APPROVED** [review]
Nice job! Let's SHIP IT!

File: Source/Serializer.swift (hunk 2 above)
Comment (joanromano) [reply Source/Serializer.swift:61]: No we need to use it from the other extensions in same file.

**Comment by MP0w** [comment]
Ci if fine, failure is usual random Xcode stuff

**Pull request merged**